codex/stores/fsstore.nim

```diff
@@ -117,27 +125,29 @@ method delBlock*(
 
   return await self.cache.delBlock(cid)
 
-method hasBlock*(self: FSStore, cid: Cid): bool =
+method hasBlock*(self: FSStore, cid: Cid): Future[?!bool] {.async.} =
   ## Check if the block exists in the blockstore
   ##
 
-  trace "Checking for block existence", cid
+  trace "Checking filestore for block existence", cid
   if cid.isEmpty:
     trace "Empty block, ignoring"
-    return true
+    return true.success
 
-  self.blockPath(cid).isFile()
+  return self.blockPath(cid).isFile().success
 
-method listBlocks*(self: FSStore, onBlock: OnBlock) {.async.} =
-  debug "Listing all blocks in store"
+method listBlocks*(self: FSStore, onBlock: OnBlock): Future[?!void] {.async.} =
```

Comment:
I'm actually not sure if `listBlock` needs a return a result, there isn't much we can do with it besides logging and moving on?

Reply:
What we will do in the case of error? Now the proposed API is unified - errors for all functions are returned encoded in Result container.

Reply:
Overall, let's discuss the proposed API in #123